packages/libs/src/sdk/utils/preparePaymentSplits.ts

```diff
@@ -0,0 +1,65 @@
+import { USDC } from '@audius/fixed-decimal'
+
+import { ExtendedPaymentSplit } from '../api/generated/default'
+import { LoggerService } from '../services'
+import { ClaimableTokensClient } from '../services/Solana/programs/ClaimableTokensClient'
+
+/**
+ * 1. Converts amounts to bigints
+ * 2. Spreads the extraAmount to each split
+ * 3. Creates user banks for receipients as necessary
+ * 4. Returns a simplified splits structure, a list of account/amount pairs
+ */
+export const prepareSplits = async ({
+  splits,
+  extraAmount,
+  claimableTokensClient,
+  logger
+}: {
+  splits: ExtendedPaymentSplit[]
+  extraAmount: bigint
+  claimableTokensClient: ClaimableTokensClient
+  logger: LoggerService
+}) => {
+  // Convert splits to big int and spread extra amount to every split
+  let amountSplits = splits.map((split, index, arr) => {
+    const amountToAdd = extraAmount / BigInt(arr.length - index)
+    extraAmount = USDC(extraAmount - amountToAdd).value
+    return {
+      ...split,
+      amount: BigInt(split.amount) + amountToAdd
+    }
+  })
+  if (extraAmount > 0) {
+    logger.debug('Calculated splits after extra amount:', amountSplits)
+  }
+
+  // Check for user banks as needed
+  amountSplits = await Promise.all(
+    amountSplits.map(async (split) => {
+      if (!split.payoutWallet) {
+        logger.debug('Deriving user bank for user...', {
+          userId: split.userId
+        })
+        const { userBank, didExist } =
+          await claimableTokensClient.getOrCreateUserBank({
```

Comment:
Minor nit: would be nice to do these in parallel as submitting and waiting for a transaction isn't a small amount of time.

Reply:
I agree generally, though in this case there's a couple things in play:
- The user bank should already exist most of the time, so this is really just covering an edge case
- The user bank needs to exist for the preflight simulation to succeed
- The "correct" way would involve putting the creation of the user bank as an instruction in the same transaction, but (1) we don't have an idempotent version of that instruction, which is slightly annoying since we'd still have to check before including it, (2) it would take away from our headroom on the purchase transaction size and (3) this seems like a cleaner side-effect with less coupling

Given that it's an edge case, I stuck with the easiest thing, but we can revisit